src/openrct2/world/Map.cpp

```diff
@@ -1184,7 +1153,7 @@ TileElement* tile_element_insert(const CoordsXYZ& loc, int32_t occupiedQuadrants
 
     // Insert new map element
     insertedElement = newTileElement;
-    newTileElement->type = 0;
+    newTileElement->SetType(static_cast<uint8_t>(type));
```

Comment:
This is not entirely the same, I think. The `type` field contains more information besides just the type and now not everything is 0-initialised.

Reply:
Good point, I'll  initialize type to 0 and call SetType afterwards.